system/include/compat/xmmintrin.h

```diff
@@ -596,8 +596,9 @@ _mm_cvtsi32_ss(__m128 __a, int __b)
 
 static __inline__ int __attribute__((__always_inline__, __nodebug__, DIAGNOSE_SLOW)) _mm_cvtss_si32(__m128 __a)
 {
-  int x = lrint(((__f32x4)__a)[0]);
-  if (x != 0 || fabsf(((__f32x4)__a)[0]) < 2.f)
+  float e = ((__f32x4)__a)[0];
+  int x = lrint(e);
+  if ((x != 0 || fabsf(e)) < 2.f && !isnan(e) && e <= (float)INT_MAX && e >= INT_MIN)
```

Comment:
Can these comparisons be made against `x` to avoid the cast here?

Is the cast needed?

Reply:
without the explicit cast, there's a warning about the implicit cast changing the value of INT_MAX.
Although looking at the other implementations in this file, they also have similar implicit casts which also produce that warning, so maybe we should be consistent.

Reply:
And we can't compare against x (at least like it's written here) because an int is always <= INT_MAX

Reply:
I found that the comparison style (e.g. whether the nan check was first or further down) was inconsistent so I made all the checks the same. I think if we want to optimize this further we should look more into using the nontrapping instructions directly but it's not clear whether that's important.